airbyte-ci/connectors/live-tests/src/live_tests/commons/connection_objects_retrieval.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+import json
+import logging
+from pathlib import Path
+from typing import Dict, Optional, Set
+
+from connection_retriever import ConnectionObject, retrieve_objects  # type: ignore
+
+from .models import Command, ConfiguredAirbyteCatalog, ConnectionObjects, SecretDict
+
+LOGGER = logging.getLogger(__name__)
+
+
+def parse_config(config: Dict | str | None) -> Optional[SecretDict]:
```

Comment:
TIL you can type objects like this in python. This looks much more legible than `Optional[Union[Dict, str]]` lol